src/app/sessions/components/details-dialog/details-dialog.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, HostListener, Inject, OnInit } from '@angular/core';
+import { DialogPosition, MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';
+import { SessionsStatisticsService } from '../../services/sessions-statistics-service';
+import * as moment from 'moment';
+
+@Component({
+  selector: 'app-details-dialog',
+  templateUrl: './details-dialog.component.html',
+  styleUrls: ['./details-dialog.component.scss']
+})
+export class DetailsDialogComponent {
```

Comment:
Additionally to my comment od SessionsStatisticsService, I can see here a mix of calculations done in the UI and service and most of them are base on Session type even though we don't know it's given in data object (not before we will look into html), as such it would be beneficial to separate this session data and related methods to another class (kind of a helper, but with a reason to 'live').
Here I have a dilemma because you could either user pipes instead of `durationAsMinutes,  parseDate, calculateEndTime, calculateStartTime` and just keep session and some calculations in a class that takes session as an argument ex. 
```
SessionDialogDetails? { 
  constructor(session, sessionsStatsService); 
  getStartTime(session) { return moment(session.start) }
  calculateAllocated() { return this.sessionsStatsService.calculateAllocated(this.data.session); }
  ...
  and so on...
}
```

so that formatting is done with pipes in html
or you can prepare that class to calculate and parse everything to what you will need in UI and provide getters or a data only type/interface` { availibility: <>, startDate: <>, startTime: <> ....}`

In general, I'm not 100% sure about given examples by me, but I don't like calculations done in the UI or relating on an unchecked type in UI, so some changes are needed here.